qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -548,4 +613,327 @@ def make_dict_next(n_qubits, dicts_prior):
+        if tpow1 > 0:
+            circuit.u1(tpow1 * np.pi / 4, 1)
+        if xpow1 == 1:
+            circuit.x(1)
+        circuit.cx(1, 0)
+        if m > 0:
+            circuit.u1(m * np.pi / 4, 0)
+
+    # CX01*CX10-like class
+    if linear == [[0, 1], [1, 1]]:
+        xpow1 = shift[0]
+        xpow0 = (shift[1] + xpow1) % 2
+        if xpow0 == xpow1:
+            m = ((8 - weight_2[0]) / 2) % 4
+            tpow0 = (weight_1[0] - m) % 8
+            tpow1 = (weight_1[1] - m) % 8
+        else:
+            m = (weight_2[0] / 2) % 4
+            tpow0 = (weight_1[0] + m) % 8
+            tpow1 = (weight_1[1] + m) % 8
+        if tpow0 > 0:
+            circuit.u1(tpow0 * np.pi / 4, 0)
+        if xpow0 == 1:
+            circuit.x(0)
+        if tpow1 > 0:
+            circuit.u1(tpow1 * np.pi / 4, 1)
+        if xpow1 == 1:
+            circuit.x(1)
+        circuit.cx(0, 1)
+        circuit.cx(1, 0)
+        if m > 0:
+            circuit.u1(m * np.pi / 4, 1)
+
+    # CX10*CX01-like class
+    if linear == [[1, 1], [1, 0]]:
+        xpow0 = shift[1]
+        xpow1 = (shift[0] + xpow0) % 2
+        if xpow0 == xpow1:
+            m = ((8 - weight_2[0]) / 2) % 4
+            tpow0 = (weight_1[0] - m) % 8
+            tpow1 = (weight_1[1] - m) % 8
+        else:
+            m = (weight_2[0] / 2) % 4
+            tpow0 = (weight_1[0] + m) % 8
+            tpow1 = (weight_1[1] + m) % 8
+        if tpow0 > 0:
+            circuit.u1(tpow0 * np.pi / 4, 0)
+        if xpow0 == 1:
+            circuit.x(0)
+        if tpow1 > 0:
+            circuit.u1(tpow1 * np.pi / 4, 1)
+        if xpow1 == 1:
+            circuit.x(1)
+        circuit.cx(1, 0)
+        circuit.cx(0, 1)
+        if m > 0:
+            circuit.u1(m * np.pi / 4, 0)
+
+    # CX01*CX10*CX01-like class
+    if linear == [[0, 1], [1, 0]]:
+        xpow0 = shift[1]
+        xpow1 = shift[0]
+        if xpow0 == xpow1:
+            m = ((8 - weight_2[0]) / 2) % 4
+            tpow0 = (weight_1[0] - m) % 8
+            tpow1 = (weight_1[1] - m) % 8
+        else:
+            m = (weight_2[0] / 2) % 4
+            tpow0 = (weight_1[0] + m) % 8
+            tpow1 = (weight_1[1] + m) % 8
+        if tpow0 > 0:
+            circuit.u1(tpow0 * np.pi / 4, 0)
+        if xpow0 == 1:
+            circuit.x(0)
+        if tpow1 > 0:
+            circuit.u1(tpow1 * np.pi / 4, 1)
+        if xpow1 == 1:
+            circuit.x(1)
+        circuit.cx(0, 1)
+        circuit.cx(1, 0)
+        if m > 0:
+            circuit.u1(m * np.pi / 4, 1)
+        circuit.cx(0, 1)
+
+    return circuit
+
+
+def random_cnotdihedral(num_qubits, seed=None):
+    """Return a random CNOTDihedral element.
+
+    Args:
+        num_qubits (int): the number of qubits for the Clifford
+        seed (int or RandomState): Optional. Set a fixed seed or
+                                   generator for RNG.
+    Returns:
+        CNOTDihedral: a random CNOTDihedral element.
+    """
+
+    np.random.seed(seed)
```

Comment:
We don't use the numpy random module anymore. See the Terra quantum info methods for examples, but this should be something like
```
if isinstance(seed, np.random.Generator):
    rng = seed
else:
   rng = np.random.default_rng(seed)
```

Reply:
Done.